src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,158 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(head, value);
+        if (tail == null) {
+            head = newNode;
+            tail = newNode;
+        } else {
+            head.next = newNode;
+            head = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        Objects.checkIndex(index,size + 1);
+        Node<T> nodeForeAdd = new Node<>(value);
+        if (tail == null) {
+            tail = head = nodeForeAdd;
+        } else if (index == 0) {
+            nodeForeAdd.next = tail;
+            tail = nodeForeAdd;
+
+        } else {
+            insert(value,index);
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
+            add(t);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        Objects.checkIndex(index,size);
+        Node<T> current = getByIndex(index);
+        return current.value;
     }
 
     @Override
-    public T set(T value, int index) {
-        return null;
+    public T set(T value,int index) {
+        return setByIndex(value,index).value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        return removeByIndex(index).value;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> current = tail;
+        for (int i = 0; i < size; i++) {
+            if (Objects.equals(current.value, object)) {
+                removeByIndex(i);
```

Comment:
instead `removeByIndex(int)` create a separate method `unlink(Node node)`. See the common mistakes https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/linked-list.html#create-a-separate-method-unlinknode-node